### ISSUE TITLE
OAuth metrics are not compatible with Cruise Control

### ISSUE THREAD

**Issue by scholzj** [strimzi/strimzi-kafka-operator#8354]
The Cruise Control metrics reporter and metrics handling in the OAuth library are not compatible with each other. When they are enabled in parallel, Kafka fails to start with errors like this:

```
2023-03-08 15:21:33,583 ERROR Exiting Kafka due to fatal exception during startup. (kafka.Kafka$) [main]
org.apache.kafka.common.KafkaException: java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap')
  at org.apache.kafka.common.network.SaslChannelBuilder.configure(SaslChannelBuilder.java:184)
  at org.apache.kafka.common.network.ChannelBuilders.create(ChannelBuilders.java:192)
  at org.apache.kafka.common.network.ChannelBuilders.serverChannelBuilder(ChannelBuilders.java:107)
  at kafka.network.Processor.<init>(SocketServer.scala:921)
  at kafka.network.Acceptor.newProcessor(SocketServer.scala:829)
  at kafka.network.Acceptor.$anonfun$addProcessors$1(SocketServer.scala:799)
  at kafka.network.Acceptor.addProcessors(SocketServer.scala:798)
  at kafka.network.DataPlaneAcceptor.configure(SocketServer.scala:502)
  at kafka.network.SocketServer.createDataPlaneAcceptorAndProcessors(SocketServer.scala:228)
  at kafka.network.SocketServer.$anonfun$new$31(SocketServer.scala:173)
  at kafka.network.SocketServer.$anonfun$new$31$adapted(SocketServer.scala:173)
  at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:575)
  at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:573)
  at scala.collection.AbstractIterable.foreach(Iterable.scala:933)
  at kafka.network.SocketServer.<init>(SocketServer.scala:173)
  at kafka.server.KafkaServer.startup(KafkaServer.scala:311)
  at kafka.Kafka$.main(Kafka.scala:109)
  at kafka.Kafka.main(Kafka.scala)
Caused by: java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap')
  at com.linkedin.kafka.cruisecontrol.metricsreporter.CruiseControlMetricsReporter.configure(CruiseControlMetricsReporter.java:188)
  at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstance(AbstractConfig.java:407)
  at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstances(AbstractConfig.java:480)
  at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstances(AbstractConfig.java:461)
  at org.apache.kafka.common.config.AbstractConfig.getConfiguredInstances(AbstractConfig.java:448)
  at io.strimzi.kafka.oauth.services.OAuthMetrics.initReporters(OAuthMetrics.java:93)
  at io.strimzi.kafka.oauth.services.OAuthMetrics.initKafkaMetrics(OAuthMetrics.java:83)
  at io.strimzi.kafka.oauth.services.OAuthMetrics.<init>(OAuthMetrics.java:63)
  at io.strimzi.kafka.oauth.services.Services.getMetrics(Services.java:72)
  at io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler.configureMetrics(JaasServerOauthValidatorCallbackHandler.java:310)
  at io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler.delegatedConfigure(JaasServerOauthValidatorCallbackHandler.java:283)
  at io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler.configure(JaasServerOauthValidatorCallbackHandler.java:229)
  at org.apache.kafka.common.network.SaslChannelBuilder.configure(SaslChannelBuilder.java:151)
```

This should be fixed, but to fix it properly, fixes in Kafka and/or Cruise Control might be required. If we are unable to solve it reasonably soon, we should consider some temporary measures:
* Document this in the documentation
* Check it in the operator and throw a warning, InvalidResourceException or just disable the OAuth metrics.

**Comment by scholzj** [comment]
Triaged on 20.4.2023: Should be kept. @mstruk is planning to look at it.

**Comment by mstruk** [comment]
This issue is addressed by #8802.
